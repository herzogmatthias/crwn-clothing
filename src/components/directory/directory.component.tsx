import * as React from "react";
import { ISection } from "../../redux/directory/ISection";
import MenuItem from "../menu-item/menu-item.components";
import { connect, ConnectedProps } from "react-redux";
import { createStructuredSelector } from "reselect";
import { RootState } from "../../redux/root-reducer";
import { selectDirectorySections } from "../../redux/directory/directory.selectors";
import { DirectoryMenuContainer } from "./directory.styles";

interface ISelectorProps {
  sections: ISection[];
}

type IDirectoryProps = ConnectedProps<typeof connector>;

function Directory({ sections }: IDirectoryProps) {
  return (
    <DirectoryMenuContainer>
      {sections.map(({ id, ...sectionProps }) => {
        return <MenuItem key={id} {...sectionProps}></MenuItem>;
      })}
    </DirectoryMenuContainer>
  );
}

const mapStateToProps = createStructuredSelector<RootState, ISelectorProps>({
  sections: selectDirectorySections,
});

const connector = connect(mapStateToProps, {});
export default connect(mapStateToProps, {})(Directory);
