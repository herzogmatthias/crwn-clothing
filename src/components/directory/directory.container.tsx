import { flowRight } from "lodash";
import { graphql } from "react-apollo";
import { ISection } from "../../graphql/ISection";
import { GET_SECTIONS } from "../../graphql/queries";
import { ApolloCurrentQueryResult } from "apollo-boost";
import Directory from "./directory.component";
import React from "react";
import { ISectionGraphql } from "../../graphql/types";

interface IWithGraphql
  extends ApolloCurrentQueryResult<{ sections: ISectionGraphql[] }> {}

const withGraphql = ({ data }: IWithGraphql) => {
  console.log(data);
  const { sections } = data!;
  return <Directory sections={sections}></Directory>;
};

export default flowRight(
  graphql<{}, { sections: ISectionGraphql[] }>(GET_SECTIONS)
)(withGraphql);
