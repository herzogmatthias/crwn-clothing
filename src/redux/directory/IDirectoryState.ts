import { ISection } from "./ISection";

export interface IDirectoryState {
  sections: ISection[];
}
