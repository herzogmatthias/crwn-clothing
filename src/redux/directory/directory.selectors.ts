import { RootState } from "../root-reducer";
import { createSelector } from "reselect";

const selectDirectory = (state: RootState) => state.directory;

export const selectDirectorySections = createSelector(
  [selectDirectory],
  (directory) => directory.sections
);
