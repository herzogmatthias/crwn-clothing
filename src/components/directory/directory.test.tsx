import React from "react";
import { ISection } from "../../redux/directory/ISection";
import {
  default as ConntectedDirectory,
  Directory,
} from "./directory.component";
import configureStore, { MockStoreEnhanced } from "redux-mock-store";
import { ReactWrapper, mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";

describe("CONNECTED DIRECTORY TESTS", () => {
  let wrapper: ReactWrapper;
  let mockStore = configureStore([]);
  let store: MockStoreEnhanced<unknown, {}>;
  const sections: ISection[] = [
    {
      id: 1,
      imageUrl: "asdasdf",
      linkUrl: "asdfasdf",
      title: "Mens",
      size: "lg",
    },
    { id: 2, imageUrl: "asdasdf", linkUrl: "asdfasdf", title: "Mens" },
  ];
  beforeEach(() => {
    const mockState = {
      directory: {
        sections: sections,
      },
    };

    store = mockStore(mockState);

    wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <ConntectedDirectory />
        </Provider>
      </BrowserRouter>
    );
  });
  test("render Directory Component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("check for props", () => {
    expect(wrapper.find(Directory).prop("sections").length).toEqual(
      sections.length
    );
  });
});
