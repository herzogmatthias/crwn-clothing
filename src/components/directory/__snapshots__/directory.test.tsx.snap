// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONNECTED DIRECTORY TESTS render Directory Component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(Directory)>
        <Directory
          sections={
            Array [
              Object {
                "id": 1,
                "imageUrl": "asdasdf",
                "linkUrl": "asdfasdf",
                "size": "lg",
                "title": "Mens",
              },
              Object {
                "id": 2,
                "imageUrl": "asdasdf",
                "linkUrl": "asdfasdf",
                "title": "Mens",
              },
            ]
          }
        >
          <styled.div>
            <div
              className="sc-AxgMl dIJrCX"
            >
              <withRouter(MenuItem)
                imageUrl="asdasdf"
                key="1"
                linkUrl="asdfasdf"
                size="lg"
                title="Mens"
              >
                <MenuItem
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  imageUrl="asdasdf"
                  linkUrl="asdfasdf"
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  size="lg"
                  title="Mens"
                >
                  <styled.div
                    onClick={[Function]}
                    size="lg"
                  >
                    <div
                      className="sc-AxjAm ciHclU"
                      onClick={[Function]}
                      size="lg"
                    >
                      <styled.div
                        url="asdasdf"
                      >
                        <div
                          className="sc-AxirZ gTjdQT"
                        />
                      </styled.div>
                      <styled.div
                        className="content"
                      >
                        <div
                          className="sc-AxiKw dlmRRm content"
                        >
                          <styled.h1
                            className="title"
                          >
                            <h1
                              className="sc-AxhCb fjAoqt title"
                            >
                              MENS
                            </h1>
                          </styled.h1>
                          <styled.span
                            className="subtitle"
                          >
                            <span
                              className="sc-AxhUy fiTrJq subtitle"
                            >
                              SHOP NOW
                            </span>
                          </styled.span>
                        </div>
                      </styled.div>
                    </div>
                  </styled.div>
                </MenuItem>
              </withRouter(MenuItem)>
              <withRouter(MenuItem)
                imageUrl="asdasdf"
                key="2"
                linkUrl="asdfasdf"
                title="Mens"
              >
                <MenuItem
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  imageUrl="asdasdf"
                  linkUrl="asdfasdf"
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  title="Mens"
                >
                  <styled.div
                    onClick={[Function]}
                  >
                    <div
                      className="sc-AxjAm ezlgHt"
                      onClick={[Function]}
                    >
                      <styled.div
                        url="asdasdf"
                      >
                        <div
                          className="sc-AxirZ gTjdQT"
                        />
                      </styled.div>
                      <styled.div
                        className="content"
                      >
                        <div
                          className="sc-AxiKw dlmRRm content"
                        >
                          <styled.h1
                            className="title"
                          >
                            <h1
                              className="sc-AxhCb fjAoqt title"
                            >
                              MENS
                            </h1>
                          </styled.h1>
                          <styled.span
                            className="subtitle"
                          >
                            <span
                              className="sc-AxhUy fiTrJq subtitle"
                            >
                              SHOP NOW
                            </span>
                          </styled.span>
                        </div>
                      </styled.div>
                    </div>
                  </styled.div>
                </MenuItem>
              </withRouter(MenuItem)>
            </div>
          </styled.div>
        </Directory>
      </Connect(Directory)>
    </Provider>
  </Router>
</BrowserRouter>
`;
